### PR TITLE
Feature/phiro

### PR DESCRIPTION
**Pull request by AdrianSch**

### CODE REVIEW

**Comment by AdrianSch** [comment]
https://jenkins.catrob.at/view/All-Categories/view/Catroid-multi-job/job/Catroid-Multi-Job-Custom-Branch-RELOADED/1863/

File: catroid/src/org/catrobat/catroid/ui/fragment/FormulaEditorFragment.java [hunk 1]
```diff
@@ -238,9 +287,33 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
 		return fragmentView;
 	}
 
+	private void handleCustomView() {
+		if (showCustomView) {
+			formulaEditorFieldDeleteButton.setVisibility(View.GONE);
+			formulaEditorEditText.setVisibility(View.GONE);
+			formulaEditorKeyboard.setVisibility(View.GONE);
+		}
+		else {
```
Comment (robertpainsi) [inline catroid/src/org/catrobat/catroid/ui/fragment/FormulaEditorFragment.java]:
```java
} else {
```

File: catroid/src/org/catrobat/catroid/ui/fragment/FormulaEditorFragment.java [hunk 2]
```diff
@@ -238,9 +287,33 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
+			formulaEditorEditText.setVisibility(View.VISIBLE);
+			formulaEditorKeyboard.setVisibility(View.VISIBLE);
+		}
+	}
+
+	private View getBrickOrCustomView() {
+		if (showCustomView) {
+			return clonedFormulaBrick.getCustomView(context, 0, null);
+		}
+		else {
```
Comment (robertpainsi) [inline catroid/src/org/catrobat/catroid/ui/fragment/FormulaEditorFragment.java]:
```java
} else {
```

File: catroid/src/org/catrobat/catroid/formulaeditor/Sensors.java [hunk 3]
```diff
@@ -27,7 +27,8 @@
 public enum Sensors {
 	X_ACCELERATION, Y_ACCELERATION, Z_ACCELERATION, COMPASS_DIRECTION, X_INCLINATION, Y_INCLINATION, LOUDNESS, FACE_DETECTED, FACE_SIZE, FACE_X_POSITION, FACE_Y_POSITION, OBJECT_X(
 			true), OBJECT_Y(true), OBJECT_TRANSPARENCY(true), OBJECT_BRIGHTNESS(true), OBJECT_SIZE(true), OBJECT_ROTATION(
-			true), OBJECT_LAYER(true), NXT_SENSOR_1, NXT_SENSOR_2, NXT_SENSOR_3, NXT_SENSOR_4;
+			true), OBJECT_LAYER(true), NXT_SENSOR_1, NXT_SENSOR_2, NXT_SENSOR_3, NXT_SENSOR_4,
+			PHIRO_FRONT_LEFT, PHIRO_FRONT_RIGHT, PHIRO_SIDE_LEFT, PHIRO_SIDE_RIGHT, PHIRO_BOTTOM_LEFT, PHIRO_BOTTOM_RIGHT;
```
Comment (robertpainsi) [inline catroid/src/org/catrobat/catroid/formulaeditor/Sensors.java]: 2 tabs to the left.
Or even better (but imo should be in it's own commit), rearrange  the upper elements, so the newline break isn't at the enum "constructor call"

File: catroid/src/org/catrobat/catroid/content/XmlHeader.java [hunk 4]
```diff
@@ -181,4 +183,8 @@ public void setScreenMode(ScreenModes screenMode) {
 	public ScreenModes getScreenMode() {
 		return this.screenMode;
 	}
+
+	boolean isPhiroProProject() { return isPhiroProProject; }
```
Comment (robertpainsi) [inline catroid/src/org/catrobat/catroid/content/XmlHeader.java]: return statement should be on its own line

File: catroid/src/org/catrobat/catroid/content/XmlHeader.java [hunk 5]
```diff
@@ -181,4 +183,8 @@ public void setScreenMode(ScreenModes screenMode) {
...
+
+	void setPhiroProProject(boolean isPhiroProProject) { this.isPhiroProProject = isPhiroProProject; }
```
Comment (robertpainsi) [inline catroid/src/org/catrobat/catroid/content/XmlHeader.java]: assignment statement should be on its own line.

**Comment by robertpainsi** [comment]
Only a few unformatted lines of **java** code according to the Android Studio formatter.
Nice!

File: catroidTest/src/org/catrobat/catroid/test/devices/phiro/PhiroImplTest.java [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+
+	private int percentToSpeed(int percent) {
+		if (percent <= 0) {
+			return 0;
+		}
+		if (percent >= 100) {
+			return 255;
+		}
+
+		return (int)(percent * 2.55);
```
Comment (robertpainsi) [inline catroidTest/src/org/catrobat/catroid/test/devices/phiro/PhiroImplTest.java]: Sy, Android Studio / I missed some files. Missing space between cast braces.
```java
return (int) (percent * 2.55);
```

File: catroidTest/src/org/catrobat/catroid/uitest/content/brick/PhiroColorBrickTest.java [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+import org.catrobat.catroid.content.bricks.Brick;
+import org.catrobat.catroid.content.bricks.PhiroRGBLightBrick;
+import org.catrobat.catroid.ui.ScriptActivity;
+import org.catrobat.catroid.ui.adapter.BrickAdapter;
+import org.catrobat.catroid.uitest.util.BaseActivityInstrumentationTestCase;
+import org.catrobat.catroid.uitest.util.UiTestUtils;
+
+import java.util.ArrayList;
+
+public class PhiroColorBrickTest extends BaseActivityInstrumentationTestCase<ScriptActivity>  {
```
Comment (robertpainsi) [inline catroidTest/src/org/catrobat/catroid/uitest/content/brick/PhiroColorBrickTest.java]: Remove one space before the opening curly brace.

File: catroidTest/src/org/catrobat/catroid/uitest/content/brick/PhiroMoveMotorBackwardBrickTest.java [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+import org.catrobat.catroid.content.bricks.Brick;
+import org.catrobat.catroid.content.bricks.PhiroMotorMoveBackwardBrick;
+import org.catrobat.catroid.ui.ScriptActivity;
+import org.catrobat.catroid.ui.adapter.BrickAdapter;
+import org.catrobat.catroid.uitest.util.BaseActivityInstrumentationTestCase;
+import org.catrobat.catroid.uitest.util.UiTestUtils;
+
+import java.util.ArrayList;
+
+public class PhiroMoveMotorBackwardBrickTest extends BaseActivityInstrumentationTestCase<ScriptActivity>  {
```
Comment (robertpainsi) [inline catroidTest/src/org/catrobat/catroid/uitest/content/brick/PhiroMoveMotorBackwardBrickTest.java]: Same as above. There are two spaces before the curly brace.

File: catroidTest/src/org/catrobat/catroid/uitest/content/brick/PhiroMoveMotorForwardBrickTest.java [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+import org.catrobat.catroid.content.bricks.Brick;
+import org.catrobat.catroid.content.bricks.PhiroMotorMoveForwardBrick;
+import org.catrobat.catroid.ui.ScriptActivity;
+import org.catrobat.catroid.ui.adapter.BrickAdapter;
+import org.catrobat.catroid.uitest.util.BaseActivityInstrumentationTestCase;
+import org.catrobat.catroid.uitest.util.UiTestUtils;
+
+import java.util.ArrayList;
+
+public class PhiroMoveMotorForwardBrickTest extends BaseActivityInstrumentationTestCase<ScriptActivity>  {
```
Comment (robertpainsi) [inline catroidTest/src/org/catrobat/catroid/uitest/content/brick/PhiroMoveMotorForwardBrickTest.java]: Same

File: catroidTest/src/org/catrobat/catroid/uitest/content/brick/PhiroPlayToneBrickTest.java [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+import org.catrobat.catroid.content.bricks.Brick;
+import org.catrobat.catroid.content.bricks.PhiroPlayToneBrick;
+import org.catrobat.catroid.ui.ScriptActivity;
+import org.catrobat.catroid.ui.adapter.BrickAdapter;
+import org.catrobat.catroid.uitest.util.BaseActivityInstrumentationTestCase;
+import org.catrobat.catroid.uitest.util.UiTestUtils;
+
+import java.util.ArrayList;
+
+public class PhiroPlayToneBrickTest extends BaseActivityInstrumentationTestCase<ScriptActivity>  {
```
Comment (robertpainsi) [inline catroidTest/src/org/catrobat/catroid/uitest/content/brick/PhiroPlayToneBrickTest.java]: Same as above.

File: catroid/res/layout/brick_phiro_rgb_light.xml [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+        android:focusable="false" >
+    <!-- do not add spinner items here! -->
+    </Spinner>
+
+    <TextView
+        android:id="@+id/brick_phiro_rgb_led_red_text_view"
+        style="@style/BrickText"
+        android:text="@string/phiro_rgb_led_red" >
+    </TextView>
+    <TextView
```
Comment (obusher) [inline catroid/res/layout/brick_phiro_rgb_light.xml]: Newline between `TextViews`.

File: catroid/src/org/catrobat/catroid/bluetooth/BluetoothDeviceFactoryImpl.java [hunk 12]
```diff
@@ -37,13 +38,17 @@
 			return new LegoNXTImpl(applicationContext);
 		}
 
-//        if (service == BTDeviceService.ALBERT) {
-//            return new Albert();
-//        }
+		if (service == BluetoothDevice.PHIRO) {
+			return new PhiroImpl();
```
Comment (obusher) [inline catroid/src/org/catrobat/catroid/bluetooth/BluetoothDeviceFactoryImpl.java]: Looks like a shortcut.

File: catroid/src/org/catrobat/catroid/content/actions/PhiroRGBLightAction.java [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+			blueValue = 0;
+			Log.d(getClass().getSimpleName(), "Formula interpretation for this specific Brick failed.", interpretationException);
+		}
+
+		if (blueValue < MIN_VALUE) {
+			blueValue = MIN_VALUE;
+		} else if (blueValue > MAX_VALUE) {
+			blueValue = MAX_VALUE;
+		}
+
```
Comment (obusher) [inline catroid/src/org/catrobat/catroid/content/actions/PhiroRGBLightAction.java]: 3 times same code -> call for a method.

File: catroid/src/org/catrobat/catroid/content/actions/PhiroSensorAction.java [hunk 14]
```diff
@@ -0,0 +1,135 @@
...
+	private boolean isInterpretedCorrectly;
+
+	protected void begin() {
+		try {
+			if (ifCondition == null) {
+				isInterpretedCorrectly = false;
+				return;
+			}
+			Double interpretation = ifCondition.interpretDouble(sprite);
+			ifConditionValue = interpretation.intValue() <= 850 ? true : false;
```
Comment (obusher) [inline catroid/src/org/catrobat/catroid/content/actions/PhiroSensorAction.java]: `850`?

File: catroid/src/org/catrobat/catroid/content/bricks/PhiroSetVariableBrick.java [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+			textSetVariable.setText(context.getResources().getString(
+					R.string.formula_editor_phiro_sensor_bottom_left));
+		} else if (this.getFormula().getRoot().getValue().equals(Sensors.PHIRO_BOTTOM_RIGHT.toString())) {
+			textSetVariable.setText(context.getResources().getString(
+					R.string.formula_editor_phiro_sensor_bottom_right));
+		} else {
+			textSetVariable.setText(String.valueOf(BrickValues.SET_VARIABLE));
+		}
+
+		return prototypeView;
```
Comment (obusher) [inline catroid/src/org/catrobat/catroid/content/bricks/PhiroSetVariableBrick.java]: http://stackoverflow.com/questions/6705955/why-switch-is-faster-than-if

**Comment by obusher** [comment]
* I would like to see a baseclass for `PhiroMotorMoveBackwardBrick/Action` and   `PhiroMotorMoveForwardBrick/Action`.
* There also should be a Catrobat Language Version incrementation.
* What about those `TODOs`?

File: build.gradle [hunk 16]
```diff
@@ -66,6 +67,8 @@ dependencies {
     androidTestCompile fileTree(include: '*.jar', dir: 'catroidTest/libs')
     androidTestCompile 'com.jayway.android.robotium:robotium-solo:5.2.1'
     pmd 'net.sourceforge.pmd:pmd:5.1.1'
+    compile files('catroid/libs/Firmata-2.6.jar')
+    compile files('catroid/libs/FirmataSerial-1.1.jar')
 }
```
Comment (obusher) [inline build.gradle]: Remove those 2 lines. [Line:65](https://github.com/Catrobat/Catroid/pull/1266/files#diff-c197962302397baf3a4cc36463dce5eaR65) should already handle compilation of your libraries. :palm_tree:

File: catroid/res/layout/brick_phiro_if_sensor.xml [hunk 17]
```diff
@@ -0,0 +1,72 @@
...
+
+        <!-- do not add spinner items here! -->
+        </Spinner >
+
+        <TextView
+            android:id="@+id/phiro_sensor_label_second_part"
+            style="@style/BrickText.SingleLine"
+            android:paddingLeft="5dip"
+            android:text="@string/brick_phiro_sensor_second_part" >
+        </TextView >
```
Comment (obusher) [inline catroid/res/layout/brick_phiro_if_sensor.xml]: I like a newline after `Textview`.

File: catroid/src/org/catrobat/catroid/content/bricks/PhiroSetVariableBrick.java [hunk 18]
```diff
@@ -0,0 +1,97 @@
...
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.common.BrickValues;
+import org.catrobat.catroid.formulaeditor.Formula;
+import org.catrobat.catroid.formulaeditor.Sensors;
+import org.catrobat.catroid.formulaeditor.UserVariable;
+import org.catrobat.catroid.ui.adapter.DataAdapter;
+import org.catrobat.catroid.ui.adapter.UserVariableAdapterWrapper;
+
+public class PhiroSetVariableBrick extends SetVariableBrick {
+
```
Comment (obusher) [inline catroid/src/org/catrobat/catroid/content/bricks/PhiroSetVariableBrick.java]: This derivation makes the `.xml` representation ambiguous.
Please modify the the prototype view in the `SetVariableBrick`.

File: catroid/src/org/catrobat/catroid/content/bricks/PhiroIfLogicBeginBrick.java [hunk 19]
```diff
@@ -0,0 +1,263 @@
...
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.content.actions.ExtendedActions;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+public class PhiroIfLogicBeginBrick extends IfLogicBeginBrick implements OnItemSelectedListener {
+	private static final long serialVersionUID = 1L;
```
Comment (obusher) [inline catroid/src/org/catrobat/catroid/content/bricks/PhiroIfLogicBeginBrick.java]: One the first glimpse this derivation looks good, but the `IfLogicBeginBrick` is a `FormulaBrick` and you
do not want that the `PhiroIfLogicBeginBrick` to be a `FormulaBrick`.
(Otherwise you get the `Edit Formula` - option in the context menu of the Brick)

File: catroid/src/org/catrobat/catroid/content/bricks/Brick.java [hunk 20]
```diff
@@ -45,7 +45,13 @@
...
 		LEGO_NXT_SPEED, LEGO_NXT_DEGREES, LEGO_NXT_FREQUENCY, LEGO_NXT_DURATION_IN_SECONDS,
 
-		DRONE_TIME_TO_FLY_IN_SECONDS, LIST_ADD_ITEM, LIST_DELETE_ITEM, INSERT_ITEM_INTO_USERLIST_VALUE, INSERT_ITEM_INTO_USERLIST_INDEX, REPLACE_ITEM_IN_USERLIST_VALUE, REPLACE_ITEM_IN_USERLIST_INDEX, DRONE_POWER_IN_PERCENT
+		DRONE_TIME_TO_FLY_IN_SECONDS, LIST_ADD_ITEM, LIST_DELETE_ITEM, INSERT_ITEM_INTO_USERLIST_VALUE,
+		INSERT_ITEM_INTO_USERLIST_INDEX, REPLACE_ITEM_IN_USERLIST_VALUE, REPLACE_ITEM_IN_USERLIST_INDEX, DRONE_POWER_IN_PERCENT,
+
+		PHIRO_SPEED, PHIRO_DURATION_IN_SECONDS, PHIRO_LIGHT_RED, PHIRO_LIGHT_GREEN, PHIRO_LIGHT_BLUE,
+		IF_PHIRO_SENSOR_CONDITION,
+
+		UNKNOWN
```
Comment (obusher) [inline catroid/src/org/catrobat/catroid/content/bricks/Brick.java]: Do we really need a unknown BrickField ?

File: catroid/src/org/catrobat/catroid/devices/arduino/phiro/PhiroImpl.java [hunk 21]
```diff
@@ -0,0 +1,384 @@
...
+import name.antonsmirnov.firmata.message.ReportAnalogPinMessage;
+import name.antonsmirnov.firmata.message.ReportFirmwareVersionMessage;
+import name.antonsmirnov.firmata.message.SetPinModeMessage;
+import name.antonsmirnov.firmata.serial.ISerial;
+import name.antonsmirnov.firmata.serial.SerialException;
+import name.antonsmirnov.firmata.serial.StreamingSerialAdapter;
+
+public class PhiroImpl implements Phiro {
+
+	private static final UUID PHIRO_UUID = UUID.fromString("00001101-0000-1000-8000-00805F9B34FB");
```
Comment (obusher) [inline catroid/src/org/catrobat/catroid/devices/arduino/phiro/PhiroImpl.java]: Looks like a unique ID which you needed in order to test your Phiro.
I assume it will not work with any other Phiro device...

File: catroid/res/layout/brick_phiro_if_sensor.xml (hunk 17 above)
Comment (aried3r) [inline catroid/res/layout/brick_phiro_if_sensor.xml]: On that note, I'd love to not have closing tags but have self closing tags like `<TextView />`.

File: catroid/res/layout/single_seekbar_value_chooser.xml [hunk 22]
```diff
@@ -0,0 +1,89 @@
...
+            android:id="@+id/single_seekbar_value"
+            style="@style/FragmentSeekbarChooserTextView"
+            android:layout_width="5dp"
+            android:layout_height="match_parent"
+            android:layout_weight="25"
+            android:text="100"
+            android:textColor="@color/white"
+            tools:ignore="HardcodedText" />
+
+    </TableRow>
```
Comment (aried3r) [inline catroid/res/layout/single_seekbar_value_chooser.xml]: Do you really need a TableLayout if all you got is a single row?

File: catroid/res/values/colors.xml [hunk 23]
```diff
@@ -75,4 +75,10 @@
     <color name="bluetooth_device_list_divider_color" >#505050</color >
     <color name="dialog_text_line" >#696969</color >
 
+    <!-- Phiro RGBColor Chooser Colors -->
+    <color name="color_chooser_rgb_red">#FFEB4618</color>
+    <color name="color_chooser_rgb_green">#FF078707</color>
+    <color name="color_chooser_rgb_blue">#FF0284e7</color>
```
Comment (aried3r) [inline catroid/res/values/colors.xml]: It's a bit redundant to declare the alpha value if it is fully opaque. Also all letters should be uppercase for consistency, there's a lowercase `e`.

File: catroid/src/org/catrobat/catroid/content/bricks/PhiroIfLogicBeginBrick.java [hunk 24]
```diff
@@ -0,0 +1,263 @@
...
+		return prototypeView;
+	}
+
+	@Override
+	public boolean isInitialized() {
+		if (ifElseBrick == null) {
+			return false;
+		} else {
+			return true;
+		}
```
Comment (aried3r) [inline catroid/src/org/catrobat/catroid/content/bricks/PhiroIfLogicBeginBrick.java]: Could be simplified to `return ifElseBrick != null;`

File: catroid/src/org/catrobat/catroid/content/bricks/PhiroIfLogicBeginBrick.java [hunk 25]
```diff
@@ -0,0 +1,263 @@
...
+	}
+
+	@Override
+	public void initialize() {
+		ifElseBrick = new IfLogicElseBrick(this);
+		ifEndBrick = new IfLogicEndBrick(ifElseBrick, this);
+		Log.w(TAG, "Creating if logic stuff");
```
Comment (aried3r) [inline catroid/src/org/catrobat/catroid/content/bricks/PhiroIfLogicBeginBrick.java]: This isn't a warning and I guess was only used during development?

File: catroid/src/org/catrobat/catroid/content/bricks/PhiroIfLogicBeginBrick.java [hunk 26]
```diff
@@ -0,0 +1,263 @@
...
+		return nestingBrickList;
+	}
+
+	@Override
+	public boolean isDraggableOver(Brick brick) {
+		if (brick == ifElseBrick) {
+			return false;
+		} else {
+			return true;
+		}
```
Comment (aried3r) [inline catroid/src/org/catrobat/catroid/content/bricks/PhiroIfLogicBeginBrick.java]: Could be simplified to `return ifElseBrick != null;`

File: catroid/src/org/catrobat/catroid/content/bricks/PhiroPlayToneBrick.java [hunk 27]
```diff
@@ -0,0 +1,238 @@
...
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.content.actions.ExtendedActions;
+import org.catrobat.catroid.formulaeditor.Formula;
+import org.catrobat.catroid.ui.fragment.FormulaEditorFragment;
+
+import java.util.List;
+
+public class PhiroPlayToneBrick extends FormulaBrick {
+
+	private static final long serialVersionUID = 1L;
```
Comment (aried3r) [inline catroid/src/org/catrobat/catroid/content/bricks/PhiroPlayToneBrick.java]: Don't think we need this.

**Comment by obusher** [comment]
I would like to see a UI-test with following flow:
* A project with `PhiroRGBLightBrick` which gets touched on the `Edittext`.
* Seekbar gets modified. 
* Click (touch) onto Edittext again to transition into Formula Editor.
* Close FE (f.e. with Back-Button)

In every step there should be appropriate asserts which check the changed Brick-Parameter.

File: catroid/src/org/catrobat/catroid/content/bricks/PhiroRGBLightBrick.java [hunk 28]
```diff
@@ -0,0 +1,335 @@
...
+import java.util.List;
+
+public class PhiroRGBLightBrick extends FormulaBrick {
+	private static final long serialVersionUID = 1L;
+
+	private transient View prototypeView;
+	private transient AdapterView<?> adapterView;
+
+	public static enum Eye {
+		Left, Right, Both
```
Comment (aried3r) [inline catroid/src/org/catrobat/catroid/content/bricks/PhiroRGBLightBrick.java]: I think we CAPITALIZE enums.

File: catroid/src/org/catrobat/catroid/devices/arduino/phiro/PhiroImpl.java [hunk 29]
```diff
@@ -0,0 +1,384 @@
...
+	@Override
+	public boolean isAlive() {
+		if (firmata == null) {
+			return false;
+		}
+
+		try {
+			firmata.send(new ReportFirmwareVersionMessage());
+			return true;
+		} catch (SerialException e) {
```
Comment (aried3r) [inline catroid/src/org/catrobat/catroid/devices/arduino/phiro/PhiroImpl.java]: @obusher, @robertpainsi, is there a checkstyle rule against exceptions named `e` or just our own code style?

File: catroid/src/org/catrobat/catroid/devices/arduino/phiro/PhiroImpl.java [hunk 30]
```diff
@@ -0,0 +1,384 @@
...
+				return phiroListener.getSideRightSensor();
+			case 1:
+				return phiroListener.getFrontRightSensor();
+			case 2:
+				return phiroListener.getBottomRightSensor();
+			case 3:
+				return phiroListener.getBottomLeftSensor();
+			case 4:
+				return phiroListener.getFrontLeftSensor();
+			case 5:
```
Comment (aried3r) [inline catroid/src/org/catrobat/catroid/devices/arduino/phiro/PhiroImpl.java]: Is there a mapping from 0..5 to the sensor or a place where this place is made explicit?

**Comment by aried3r** [comment]
I've seen quite a few formatting errors, instead of pointing them all out I'd ask you to run the formatter on your code.

File: catroid/src/org/catrobat/catroid/content/bricks/PhiroIfLogicBeginBrick.java (hunk 24 above)
Comment (robertpainsi) [inline catroid/src/org/catrobat/catroid/content/bricks/PhiroIfLogicBeginBrick.java]: A checkstyle rule is waiting to handle such simplifications. :smirk_cat:

However, the class is a whole c&p of the base class IfLogicBeginBrick and many of these methods shouldn't be overwritten. Or at least not overwritten, if the code is exactly the same. Also some members don't have to be redeclared the same way as in the base class. (e.g. ifElseBrick or iEndBrick)

**Comment by robertpainsi** [comment]
@aried3r: You are right, there are still some formatting errors in the code. Don't know why I missed them :pouting_cat:

@AdrianSch: Using this guide, you should be able to fix all errors at once: https://gist.github.com/robertpainsi/bf86fc95bb9e72493e83

All xml changes during formatting has to be inspected manually because of https://jira.catrob.at/browse/CAT-1406

File: catroid/src/org/catrobat/catroid/devices/arduino/phiro/PhiroImpl.java (hunk 29 above)
Comment (robertpainsi) [inline catroid/src/org/catrobat/catroid/devices/arduino/phiro/PhiroImpl.java]: Work in progress: [CAT-1277#comment1](https://jira.catrob.at/browse/CAT-1277?focusedCommentId=16348&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16348)
(Since you may find the naming convention a little tight, also consider to read [CAT-1277#comment2](https://jira.catrob.at/browse/CAT-1277?focusedCommentId=16354&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16354))

File: catroid/src/org/catrobat/catroid/devices/arduino/phiro/PhiroImpl.java (hunk 21 above)
Comment (martinbu) [inline catroid/src/org/catrobat/catroid/devices/arduino/phiro/PhiroImpl.java]: No, it is a bluetooth uuid. This is the default one. Also Lego, Arduino, Albert, etc. use it.

Comment (obusher) [inline catroid/src/org/catrobat/catroid/devices/arduino/phiro/PhiroImpl.java]: Ah, nice to know! Would be nice if you could rename the constant to something like `DEFAULT_BLUETOOTH_UUID`.

**Comment by robertpainsi** [comment]
Before merging the pull request, feel free to squash all code review commits into a single one. :smiley_cat:
